### PR TITLE
Move CI to GitHub Actions

### LINKED ISSUE #94: Move CI builds to use github workflows
No point using a third party anymore when we can just have direct integration here IMO.

Would be good to get things move off travis onto a GH workflow

### PR DESCRIPTION
**Pull request by stramel**
Fixes #94

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: Build & Test
+
+on:
+  push:
+    branches: [main]
```
Comment (43081j) [inline .github/workflows/main.yml:5]: this one too should be `master` i think

Comment (stramel) [reply .github/workflows/main.yml:5]: whoops lol copied from my work scheduling

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+name: Build & Test
+
+on:
+  push:
+    branches: [master]
+  pull_request:
```
Comment (43081j) [inline .github/workflows/main.yml:6]: also does this mean it'll run on every PR? or do we need some kind of wildcard here? just so i understand

Comment (stramel) [reply .github/workflows/main.yml:6]: Yeah this runs on every PR. It basically says "on PR"

**Pull request merged**